app/src/main/java/com/github/siela1915/bootcamp/AutocompleteApi/IngredientAutocomplete.java

```diff
@@ -94,38 +110,84 @@ public void onFailure(Call<List<ApiResponse>> call, Throwable t) {
         return ingredients;
     }
 
-    public void getNutritionFromIngredient(int id, Ingredient ingredient){
+    public void getNutritionFromIngredient(int id, Ingredient ingredient, Map<Integer, Boolean> finishedMap, UploadCallback callback){
+        //System.out.println(service.getNutrition(id, ingredient.getUnit().getValue(), API_KEY).request().url().toString());
+
+        //To be able to get the nutrition correctly for a given unit, we need a naming convention for the units which I am not sure how to do
+        //TODO give the users possible options of units once he chooses an ingredient?
         service.getNutrition(id, ingredient.getUnit().getValue(), ingredient.getUnit().getInfo(), API_KEY).enqueue(new Callback<NutrientsResponse>() {
             @Override
             public void onResponse(Call<NutrientsResponse> call, Response<NutrientsResponse> response) {
                 if(response.isSuccessful()) {
                     if (response.body() != null) {
-                        for (Nutrient nut : response.body().nutrients) {
-                            switch(nut.name){
+                        Map<String, Double> map = response.body().nutrition.mapFromNutrients();
+                        for(String s : Nutrition.NUTRIENT_NAMES){
+                            switch (s) {
+                                //map.get() won't return null due to how the map is constructed in the Nutrition class
                                 case "Calories":
-                                    ingredient.setCalories(nut.amount);
+                                    ingredient.setCalories(map.get(s));
                                     break;
                                 case "Fat":
-                                    ingredient.setFat(nut.amount);
-                                    break;
-                                case "Carbohydrates":
-                                    ingredient.setCarbs(nut.amount);
+                                    ingredient.setFat(map.get(s));
                                     break;
                                 case "Sugar":
-                                    ingredient.setSugar(nut.amount);
+                                    ingredient.setSugar(map.get(s));
                                     break;
                                 case "Protein":
-                                    ingredient.setProtein(nut.amount);
+                                    ingredient.setProtein(map.get(s));
+                                    break;
+                                case "Carbohydrates":
+                                    ingredient.setCarbs(map.get(s));
+                                    break;
+                                default:
```

Comment:
You might want to throw an exception in the default case in case something goes wrong in the construction of the nutrients.